### PR TITLE
remove configloader helper objects

### LINKED ISSUE #154: Update docs to reflect metrics migration
[Docs like these](https://mozilla.github.io/mozanalysis/api/metrics/desktop.html) are empty even though the namespaces are not.

### PR DESCRIPTION
**Pull request by jaredsnyder**
Fixes #154

Removes deprecated metrics and segments subpackages.  Important classes definied in the `__init__.py` of these were moved to `modules.py` and `segments.py` files.

I haven't worked with sphinx much so the main thing I'm worried about is that I broke it by deleting those `.rst` files

### CODE REVIEW

File: tests/test_experiment.py [hunk 1]
```diff
@@ -17,6 +12,44 @@
 from mozanalysis.segments import Segment, SegmentDataSource
 
 
+@pytest.fixture()
+def desktop_metrics():
```
Comment (danielkberry) [inline tests/test_experiment.py:16]: Seems like this functionality is defined in a bunch of test files. How about placing it in a single file and then all the test files can import?

File: tests/test_experiment.py [hunk 2]
```diff
@@ -319,15 +352,22 @@ def test_megaquery_not_detectably_malformed():
...
 
-def test_segments_megaquery_not_detectably_malformed():
+def test_segments_megaquery_not_detectably_malformed(desktop_metrics):
+    desktop_segments = [
+        ConfigLoader.get_segment("regular_users_v3", "firefox_desktop"),
+        ConfigLoader.get_segment("new_or_resurrected_v3", "firefox_desktop"),
+        ConfigLoader.get_segment("weekday_regular_v1", "firefox_desktop"),
+        ConfigLoader.get_segment("allweek_regular_v1", "firefox_desktop"),
+        ConfigLoader.get_segment("new_unique_profiles", "firefox_desktop"),
+    ]
```
Comment (danielkberry) [inline tests/test_experiment.py:370]: Same question here: can we define test segments in that single file?

File: tests/test_experiment.py [hunk 3]
```diff
@@ -339,22 +379,22 @@ def test_segments_megaquery_not_detectably_malformed():
 
     enrollments_sql = exp.build_enrollments_query(
         time_limits=tl,
-        segment_list=[s for s in msd.__dict__.values() if isinstance(s, Segment)],
+        segment_list=desktop_segments,
```
Comment (danielkberry) [inline tests/test_experiment.py:382]: The new approach seems much cleaner

File: tests/test_metric_libraries.py [hunk 4]
```diff
@@ -1,51 +1,102 @@
...
+        other_datasources += [
+            ConfigLoader.get_data_source("baseline", platform),
+            ConfigLoader.get_data_source("events", platform),
+            ConfigLoader.get_data_source("metrics", platform),
+        ]
+
+    return desktop_datasources + other_datasources
 
 
 def test_sql_not_detectably_malformed(included_metrics, included_datasources):
```
Comment (danielkberry) [inline tests/test_metric_libraries.py:91]: Does this test mean that mozanalysis is testing the validity of definitions from metric-hub? Or is this testing something about mozanalysis's sql generation? I'd think the former should be the responsibility of metric-hub to test, though they're only testing on merges, not continuously.

**Review by danielkberry — CHANGES_REQUESTED** [review]
Overall good, my biggest comment is that there's a lot of lists of segments/metrics/data sources being defined throughout the tests. If possible, I think we should simplify that into one place and import throughout the tests.

File: tests/test_metric_libraries.py (hunk 4 above)
Comment (jaredsnyder) [reply tests/test_metric_libraries.py:91]: Yeah I was wondering about that.  These tests are essentially testing the functionality of configloader, so the question is whether or not these give us anything the tests in `test_config` don't?

Comment (jaredsnyder) [reply tests/test_metric_libraries.py:91]: And if they are doing something useful we should maybe just put them in `test_config.py`

Comment (jaredsnyder) [reply tests/test_metric_libraries.py:91]: Looking it over all this is covered in `test_config.py`, I vote we delete `test_metric_librarires.py` and `test_segement_libraries.py`

File: tests/test_experiment.py (hunk 1 above)
Comment (jaredsnyder) [reply tests/test_experiment.py]: Added in `helpers/config_loader_lists.py`

File: tests/test_experiment.py (hunk 2 above)
Comment (jaredsnyder) [reply tests/test_experiment.py]: Added in helpers/config_loader_lists.py

**Review by danielkberry — APPROVED** [review]
LGTM

**Pull request merged**